lib/blocks.php

```diff
@@ -461,3 +461,25 @@ function gutenberg_should_render_lightbox( $block ) {
 }
 
 add_filter( 'render_block_data', 'gutenberg_should_render_lightbox', 15, 1 );
+
+/**
+ * Registers the metadata block attribute for all block types.
+ *
+ * @param array $args Array of arguments for registering a block type.
+ * @return array $args
+ */
+function gutenberg_register_metadata_attribute( $args ) {
```

Comment:
Specifically @ntsekouras this is where it is being promoted to non-experimental. This doesn't actually change much other than it's no longer in the `experimental` directory.

Reply:
I think having the `metadata` support is okay and has been discussed in many other issues, so it could be helpful. Should we add it in a different php file under `block-supports` folder though?

Reply:
Yeh I was wondering that. However this is no longer a block _supports_ is it? It's just a feature of every block. 

If so then maybe it should be somewhere else in a more fundamental place lower down in the code?

Reply:
@ntsekouras I wonder whether infact it would be better to register it as a global attribute available to every block such as seems to have been done for `lock`.

https://github.com/WordPress/wordpress-develop/blob/a80165b78f2367f92e2eff679c4e035d1c5bdc7c/src/wp-includes/class-wp-block-type.php#L248-L250

Maybe @Mamaduka can shed some light on the best approach here? 🙏

Reply:
Global attributes seems a good place for this, yes.

Reply:
Sorry, missed the ping.

If the attribute should be available for every block, then using a global attribute makes sense.

Here's the core PR: https://github.com/WordPress/wordpress-develop/pull/2615, if you're looking for more context.

P.S. I think we can drop the `block` prefix from the support flag and user `renaming`. I would double-check the other flags for consistency.

Reply:
@ockham It looks like you backported this to Core in https://core.trac.wordpress.org/changeset/57068.

I was going to say we can remove this code but would that cause backwards compat with versions of WP Core which don't have the backported code but are running the Plugin?

If so it seems to be gated code:

`if ( ! array_key_exists( 'metadata', $args['attributes'] ) ) {`

...so we should probably move to a `compat/wordpress-*` directory? What do you think?

Reply:
@getdave I think that makes sense! `compat/wordpress-6.5/` sounds like the right place for this 👍

Reply:
I've been looking everywhere for this comment 😅

Reply:
https://github.com/WordPress/gutenberg/pull/58126